modulemd/include/modulemd-1.0/private/modulemd-yaml.h

```diff
@@ -75,46 +75,50 @@ typedef gboolean (*ModulemdParsingFunc) (yaml_parser_t *parser,
+      g_debug (__VA_ARGS__);                                                  \
+      g_set_error (_error, MODULEMD_YAML_ERROR, type, __VA_ARGS__);           \
       result = FALSE;                                                         \
       goto error;                                                             \
     }                                                                         \
   while (0)
 
-#define MMD_YAML_ERROR_RETURN(_error, msg)                                    \
+#define MMD_YAML_ERROR_RETURN(_error, ...)                                    \
   do                                                                          \
     {                                                                         \
-      g_debug (msg);                                                          \
-      g_set_error_literal (                                                   \
-        _error, MODULEMD_YAML_ERROR, MODULEMD_YAML_ERROR_PARSE, msg);         \
+      g_debug (__VA_ARGS__);                                                  \
+      g_set_error (                                                           \
+        _error, MODULEMD_YAML_ERROR, MODULEMD_YAML_ERROR_PARSE, __VA_ARGS__); \
       result = FALSE;                                                         \
       goto error;                                                             \
     }                                                                         \
   while (0)
 
-#define YAML_EMITTER_EMIT_WITH_ERROR_RETURN(emitter, event, _error, msg)      \
+#define YAML_EMITTER_EMIT_WITH_ERROR_RETURN(emitter, event, _error, ...)      \
   do                                                                          \
     {                                                                         \
       if (!yaml_emitter_emit (emitter, event))                                \
         {                                                                     \
+          gchar *formatted = g_strdup_printf (__VA_ARGS__);                   \
           g_debug ("Error: %s - event type: %s",                              \
-                   msg,                                                       \
+                   formatted,                                                 \
                    mmd_yaml_get_event_name ((event)->type));                  \
-          g_set_error_literal (                                               \
-            _error, MODULEMD_YAML_ERROR, MODULEMD_YAML_ERROR_EMIT, msg);      \
+          g_set_error_literal (_error,                                        \
+                               MODULEMD_YAML_ERROR,                           \
+                               MODULEMD_YAML_ERROR_EMIT,                      \
+                               formatted);                                    \
+          g_free (formatted);                                                 \
```

Comment:
If you make the `g_autofree` change above, this line becomes unnecessary.

Reply:
I wasn't aware of `g_autofree()`, neat trick...